lib/datasource/sbt-plugin/index.ts

```diff
@@ -78,32 +79,43 @@ export async function getReleases({
   const [artifact, scalaVersion] = artifactIdSplit;
 
   const repoRoot = ensureTrailingSlash(registryUrl);
-  const searchRoots: string[] = [];
+  const groupIds: string[] = [];
   // Optimize lookup order
-  searchRoots.push(`${repoRoot}${groupIdSplit.join('.')}`);
-  searchRoots.push(`${repoRoot}${groupIdSplit.join('/')}`);
+  groupIds.push(`${groupIdSplit.join('/')}`);
+  groupIds.push(`${groupIdSplit.join('.')}`);
```

Comment:
Main motivation for this change is to be able to pass the `repoRoot` since maven utils is working with that.
One could argue that if this is a real problem the logic should also move into maven utils to be able to resolve this.
I'm not sure if this happens more frequently for scala packages - however usually those are considered "wrong". 

Example: 
https://repo.maven.apache.org/maven2/org.elasticsearch.client/rest/5.0.0-beta1/
however also 
https://repo.maven.apache.org/maven2/org/elasticsearch/client/rest/5.0.0-beta1/
exists - but we cannot be sure how self-hosted maven repositories look like.